// @langchain/pinecone/dist/translator.cjs
require("./_virtual/_rolldown/runtime.cjs");
let _langchain_core_structured_query = require("@langchain/core/structured_query");
//#region src/translator.ts
/**
* Specialized translator class that extends the BasicTranslator. It is
* designed to work with PineconeStore, a type of vector store in
* LangChain. The class is initialized with a set of allowed operators and
* comparators, which are used in the translation process to construct
* queries and compare results.
* @example
* ```typescript
* const selfQueryRetriever = SelfQueryRetriever.fromLLM({
*   llm: new ChatOpenAI({ model: "gpt-4o-mini" }),
*   vectorStore: new PineconeStore(),
*   documentContents: "Brief summary of a movie",
*   attributeInfo: [],
*   structuredQueryTranslator: new PineconeTranslator(),
* });
*
* const queryResult = await selfQueryRetriever.getRelevantDocuments(
*   "Which movies are directed by Greta Gerwig?",
* );
* ```
*/
var PineconeTranslator = class extends _langchain_core_structured_query.BasicTranslator {
	constructor() {
		super({
			allowedOperators: [_langchain_core_structured_query.Operators.and, _langchain_core_structured_query.Operators.or],
			allowedComparators: [
				_langchain_core_structured_query.Comparators.eq,
				_langchain_core_structured_query.Comparators.ne,
				_langchain_core_structured_query.Comparators.gt,
				_langchain_core_structured_query.Comparators.gte,
				_langchain_core_structured_query.Comparators.lt,
				_langchain_core_structured_query.Comparators.lte
			]
		});
	}
};
//#endregion
exports.PineconeTranslator = PineconeTranslator;
